File: xssfinder-java-executor/src/main/java/org/xssfinder/xss/XssGenerator.java
package org.xssfinder.xss;

/**
 * Generates uniquely identifiable XSS attacks
 */
public class XssGenerator {
    private final XssAttackFactory xssAttackFactory;
    private int nextXssId = 1;

    public XssGenerator(XssAttackFactory xssAttackFactory) {
        this.xssAttackFactory = xssAttackFactory;
    }

    public XssAttack createXssAttack() {
        XssAttack attack = xssAttackFactory.createXssAttack(Integer.toString(nextXssId));
        nextXssId++;
        return attack;
    }
}

File: xssfinder-php-executor/src/XssFinder/Xss/XssGenerator.php
<?php

namespace XssFinder\Xss;

class XssGenerator
{
    /** @var XssAttackFactory */
    private $_attackFactory;
    /** @var int */
    private $_nextXssId;

    public function __construct(XssAttackFactory $attackFactory)
    {
        $this->_attackFactory = $attackFactory;
        $this->_nextXssId = 1;
    }

    public function createXssAttack()
    {
        $id = (string)$this->_nextXssId;
        $this->_nextXssId++;
        return $this->_attackFactory->createXssAttack($id);
    }
}